app/models/auth_sources/auth_source_ldap.rb

```diff
@@ -130,6 +131,28 @@ def users_in_group(name)
     ldap_con.user_list(name)
   end
 
+  def test_connection
+    result = {}
+    begin
+      Timeout::timeout(10) do
+        self.ldap_con.valid_user?("")
```

Comment:
This test method is not complete.
For example, if you enter wrong  account user/password, it will return true.
So I chose valid_user for more accurate connection testing.